### ISSUE TITLE
[Question] What are the plans for Teleporters and Mag-Locks?

### ISSUE THREAD

**Issue by Crimson-Artist** [tsunamayo/Starship-EVO#5645]
Teleporters and Mag-Locks are both very powerful and game changing systems however right now they lack any kind of balancing. I have yet to hear any indication of how they will be balanced for survival so I guess nows a better time than any to ask. Let me list my concerns for each.
Teleporters

- There are no restrictions on them at the moment. Presumably anyone can teleport to a pad regardless of faction or hostility. Boarding parties could just teleport into your ship via a pad.
- They draw little to no power. For something as instant as teleportation there needs to be a much larger power draw. Adding the ability to turn it off with logic should help with power management.
- Pads have a orientation but there is no arrow indicating it. When you teleport to a pad the direction you face is determined by the pads orientation.
- Pads are omni-directional. Having a linked version of teleport pads would help with securing certain transportation routes.
- the max range is shockingly low compared to how large the universe is. Its max range is 10km and this severely restricts its usefulness.

Mag-Locks

- No restrictions on what can and can not be locked. I've mag-locked anchored stations and moved them around with ease. Enemy ships can be locked mid-combat to render them helpless.
- The mass limit mentioned in the description is not implemented. Presumably the mass limit will depend on the scale of the mag-lock.
- Functionally serves the same purpose as Shipyards at the moment. Objects mag-locked can be brought along in lightcruise and warp.
- A reverse mag-lock options should be added. In other words the option to set a mag-lock to attach its parent entity to something else. Think boarding pods using reverse mag-lock to attach themselves to the hull of a ship.

**Comment by CoasterBlock** [comment]
For teleporters id like some kind of "adress" and "public/private" mode. Similar to some devices in modded minecraft. 
My idea is to allow teleportation in a certain range. Lets say 1000m around the teleporter. But only with both grids have power, are in the range, the teleporters are part of the some network (owned by the player/faction)

**Comment by Tackietacktack** [comment]
Yes, and it will be nice if we can build more teleporter for one entity.
With this, we can build more teleporter for one great ship, now we must use connector to get a new entity.
If every teleporter get a personal ID like "teleporter 1","teleporter 2" or "hangar", "cockpit"  than we can travel better on great ship or station.

**Comment by tsunamayo** [comment]
@Crimson-Artist Hi, yes both needs a solid rework.
